libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -69,14 +72,18 @@ export class CartService {
     // - Fetch stripeCustomerId if uid is passed and has a customer id
     let accountCustomer;
     if (args.uid) {
-      try {
-        accountCustomer =
-          await this.accountCustomerManager.getAccountCustomerByUid(args.uid);
-      } catch (error) {
-        if (!(error instanceof AccountCustomerNotFoundError)) {
-          throw error;
-        }
-      }
+      let _;
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      [accountCustomer, _] = await Promise.all([
+        this.accountCustomerManager
+          .getAccountCustomerByUid(args.uid)
+          .catch((error) => {
+            if (!(error instanceof AccountCustomerNotFoundError)) {
+              throw error;
+            }
+          }),
+        this.updateSentryForAccount(args.uid),
```

Comment:
It would evaluate every one yes, but would set the flag according to the user's `metricsOptOutAt` db field. If the user hasn't opted out of metrics, then the `metricsOptOut` sentry flag gets set to `false`, and the metrics get logged as expected